CleverGirl/CleverGirl/Helper/BehaviorHelper.cs

```diff
@@ -17,23 +20,43 @@ public static BehaviorVariableValue GetBehaviorVariableValue(BehaviorTree bTree,
             {
                 // Ask RolePlayer for the variable
                 //getBehaviourVariable(AbstractActor actor, BehaviorVariableName name)
-                Mod.Log.Trace?.Write($"Pulling BehaviorVariableValue from RolePlayer for unit: {bTree.unit.DistinctId()}.");
+                Mod.Log.Trace?.Write($"Pulling BehaviorVariableValue {name} from RolePlayer for unit: {bTree.unit.DistinctId()}.");
                 bhVarVal = (BehaviorVariableValue)ModState.RolePlayerGetBehaviorVar.Invoke(ModState.RolePlayerBehaviorVarManager, new object[] { bTree.unit, name });
             }
 
             if (bhVarVal == null)
             {
                 // RolePlayer does not return the vanilla value if there's no configuration for the actor. We need to check that we're null here to trap that edge case.
                 // Also, if RolePlayer isn't configured we need to read the value. 
-                Mod.Log.Trace?.Write($"Pulling BehaviorVariableValue from Vanilla for unit: {bTree.unit.DistinctId()}.");
+                Mod.Log.Trace?.Write($"Pulling BehaviorVariableValue {name} from Vanilla for unit: {bTree.unit.DistinctId()}.");
                 bhVarVal = GetBehaviorVariableValueDirectly(bTree, name);
             }
 
-            Mod.Log.Trace?.Write($"  Value is: {bhVarVal}");
+            Mod.Log.Trace?.Write($" Value of {name} with type {bhVarVal.type} is {bhVarVal.GetValueString()}");
             return bhVarVal;
 
         }
 
+        private static string GetValueString(this BehaviorVariableValue behaviorVariableValue)
+        {
+            switch (behaviorVariableValue.type)
+            {
+                case BehaviorVariableValue.BehaviorVariableType.Undefined:
+                case BehaviorVariableValue.BehaviorVariableType.EncounterObjectGameLogic:
+                    return "Undefined";
+                case BehaviorVariableValue.BehaviorVariableType.Float:
+                    return behaviorVariableValue.FloatVal.ToString();
+                case BehaviorVariableValue.BehaviorVariableType.Int:
+                    return behaviorVariableValue.IntVal.ToString();
+                case BehaviorVariableValue.BehaviorVariableType.Bool:
+                    return behaviorVariableValue.BoolVal.ToString();
+                case BehaviorVariableValue.BehaviorVariableType.String:
+                    return behaviorVariableValue.StringVal;
+                default:
+                    throw new ArgumentOutOfRangeException();
```

Comment:
Throwing an exception here will cause the entire logic to brick (because it's not caught elsewhere), which I think should cause an AI unit to brace. If that's intentional great, but might be better to return an empty string?